packages/amplify-e2e-tests/src/init/amplifyPush.ts

```diff
@@ -1,14 +1,15 @@
 import { nspawn as spawn } from 'amplify-e2e-core';
-import { getCLIPath } from '../utils';
+import { getCLIPath, getProjectMeta, describeCloudFormationStack } from '../utils';
+import { retry } from '../utils/retrier';
 
-export function amplifyPush(cwd: string) {
-  return new Promise((resolve, reject) => {
+export const amplifyPush = async (cwd: string) => {
+  const { StackName: stackName, Region: region } = getProjectMeta(cwd).providers.awscloudformation;
+  await new Promise((resolve, reject) => {
     spawn(getCLIPath(), ['push'], { cwd, stripColors: true })
```

Comment:
Could you increase the test timeout to 5 minutes using `noOutputTimeout`. I have seen CFN not sending anything in 2 minutes

Reply:
That's effectively what I've done because I took out the wait in the nexpect flow and added a wait on the stack status (and the default wait there is 5 minutes). That should be better because it will allow us to fail fast in cases where we are expecting the wrong thing in the CLI flow, but still allows us to wait for pushes

Reply:
When the nexpect times out it will kill the process before exiting. So your stack would neither be updated nor subsequent tests would be run. The fix that you have would help us in for the afterEach step where we tear down all the resources

Reply:
But nexpect won't timeout becuase it will just send the push command, then 'y', then as far as it's concerned, it's done. But then we poll the stack status to determine when the push is actually complete.

Reply:
increased the nexpect timeout for push to 10 mins and moved the await CFN complete to the top of the deleteProject() function to avoid trying to delete before the updates are complete.